diff --git a/browser/base/content/tabbrowser.xml b/browser/base/content/tabbrowser.xml
index b7f829c..73d38dd 100644
--- a/browser/base/content/tabbrowser.xml
+++ b/browser/base/content/tabbrowser.xml
@@ -1593,6 +1593,13 @@
 
             this.tabContainer.appendChild(t);
 
+            //Start of move tab patch =eva
+            //get index of the current tab
+            var newIndex;
+            oldIndex = this.mTabContainer.selectedIndex;
+            //the new index will be the one on the left
+            newIndex = oldIndex - 1;
+
             // If this new tab is owned by another, assert that relationship
             if (aOwner)
               t.owner = aOwner;
@@ -1705,6 +1712,9 @@
             evt.initEvent("TabOpen", true, false);
             t.dispatchEvent(evt);
 
+            //Making the move of the tab at the end of method =eva
+            this.moveTabTo(t, newIndex);
+
             // If we didn't swap docShells with a preloaded browser
             // then let's just continue loading the page normally.
             if (!docShellsSwapped && !uriIsAboutBlank) {
diff --git a/browser/locales/en-US/chrome/browser/browser.dtd b/browser/locales/en-US/chrome/browser/browser.dtd
index 1d9c509..2f959d4 100644
--- a/browser/locales/en-US/chrome/browser/browser.dtd
+++ b/browser/locales/en-US/chrome/browser/browser.dtd
@@ -166,13 +166,13 @@ These should match what Safari and other Apple applications use on OS X Lion. --
 <!ENTITY backForwardButtonMenu.tooltip "Right-click or pull down to show history">
 <!ENTITY backForwardButtonMenuMac.tooltip "Pull down to show history">
 <!ENTITY reloadCmd.label              "Reload">
-<!ENTITY reloadButton.tooltip         "Reload current page">
-<!ENTITY stopCmd.label                "Stop">
+<!ENTITY reloadButton.tooltip         "Clap your hands!">
+<!ENTITY stopCmd.label                "Stop, in the name of love!">
 <!ENTITY stopCmd.macCommandKey        ".">
 <!ENTITY stopButton.tooltip           "Stop loading this page">
 <!ENTITY goEndCap.tooltip             "Go to the address in the Location Bar">
 <!ENTITY printButton.label            "Print">
-<!ENTITY printButton.tooltip          "Print this page">
+<!ENTITY printButton.tooltip          "Do you really want to print me?">
 
 <!ENTITY locationItem.title           "Location">
 <!ENTITY searchItem.title             "Search">
